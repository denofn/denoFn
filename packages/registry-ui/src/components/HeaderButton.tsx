import { css } from "otion";
import React from "react";

export type ButtonProps = {
  title: string;
  onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
};

export function HeaderButton({ title, onClick }: ButtonProps) {
  return (
    <button
      onClick={onClick}
      className={css({
        boxSizing: "border-box",
        backgroundColor: "transparent",
        color: "white",
        borderColor: "white",
        fontWeight: "bold",
        borderWidth: "1px",
        borderStyle: "solid",
        borderRadius: "0.25rem",
        padding: "7px 1rem",
        cursor: "pointer",
        lineHeight: "1rem",
        fontSize: "1rem",
        ":hover": {
          color: "black",
          backgroundColor: "white",
          borderColor: "transparent",
        },
      })}
    >
      {title}
    </button>
  );
}
