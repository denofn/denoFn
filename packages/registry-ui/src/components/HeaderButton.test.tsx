import { render, cleanup } from "@testing-library/react";
import React from "react";

import { HeaderButton } from "./HeaderButton";

afterEach(cleanup);

it("renders with the correct styles", () => {
  const { asFragment } = render(<HeaderButton title="title" onClick={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
