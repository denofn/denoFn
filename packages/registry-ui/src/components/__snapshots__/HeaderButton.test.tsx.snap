// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with the correct styles 1`] = `
<DocumentFragment>
  <button
    class="_1khn195 _1peqg37 _1y5pc60 _1oscket _1efi8gv _my9kfw _fxc8cu _zowvxu _luq9nz _e0dnmk _r24itg _rkiwex _807wit _16qj0cz _ryz482"
  >
    title
  </button>
</DocumentFragment>
`;
